### PR TITLE
Add support for Neviweb v2

### LINKED ISSUE #5: Unhandled rejection TypeError / new neviweb version
looks like the new neviweb update broke homebridge plugin compatibility. 
getting the below error in the logs now:
`info: Outgoing GET https://neviweb.com/api/gateway
Unhandled rejection TypeError: Cannot read property 'id' of undefined
    at prop (/usr/local/lib/node_modules/homebridge-sinope/node_modules/ramda/dist/ramda.js:3399:19)
    at /usr/local/lib/node_modules/homebridge-sinope/node_modules/ramda/dist/ramda.js:466:28
    at f1 (/usr/local/lib/node_modules/homebridge-sinope/node_modules/ramda/dist/ramda.js:446:27)
    at /usr/local/lib/node_modules/homebridge-sinope/node_modules/ramda/dist/ramda.js:296:22
    at /usr/local/lib/node_modules/homebridge-sinope/node_modules/ramda/dist/ramda.js:48:27
    at request.then.gateway (/usr/local/lib/node_modules/homebridge-sinope/src/kit/accessory.factory.js:17:22)
    at tryCatcher (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/usr/local/lib/node_modules/homebridge-sinope/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:756:18)
    at tryOnImmediate (timers.js:717:5)`

### PR DESCRIPTION
**Pull request by palourde**
This PR closes https://github.com/jfdoube/homebridge-sinope/issues/5

It essentially adds support for the new v2 API of Neviweb. I think it should still be considered a WIP, I'll be running this patch for the next couple of days but I'd definitely appreciate if other users could try it out.

I apologize in advance, I'm not a JavaScript developer so I had to hack my way around 😬

### CODE REVIEW

File: src/kit/sinope.platform.js [hunk 1]
```diff
@@ -68,4 +73,21 @@ function SinopeThermostatAccessory(log, services) {
...
+function logout() {
+  request({
+    method: 'GET',
+    path: ['logout'],
+    sessionId: session,
+  })
+  .then(response => {
+    if (response.success === true) {
+      console.log('successfully logged out')
+      return
```
Comment (jfdoube) [inline src/kit/sinope.platform.js]: No need for a return. It will already return right after the if.

File: src/kit/accessory.factory.js [hunk 2]
```diff
@@ -5,27 +5,32 @@ function accessories(log, accessoryMaker, loginPromise) {
   return (homeBridgeCallback) => {
 
     loginPromise.then(authData => {
-      const { session } = authData;
+      session = authData.session
```
Comment (jfdoube) [inline src/kit/accessory.factory.js:8]: `const { session } = authData` achieve the samething.

File: src/kit/accessory.factory.js [hunk 3]
```diff
@@ -5,27 +5,32 @@ function accessories(log, accessoryMaker, loginPromise) {
...
+          path: ['devices'],
           sessionId: session,
           queryString,
         });
       })
       .then(devices => {
-        R.call(homeBridgeCallback, R.map(accessoryMaker, devices));
+        // Gateways are now returned in the list of devices so we need to
+        // filter these out. 
+        const thermostats = devices.filter(function(device) {
```
Comment (jfdoube) [inline src/kit/accessory.factory.js]: on a simpler way of writing it,
```
...
const thermostats = devices.filter(device => device['parentDevice$id'] !== null)
...
```

**Comment by jfdoube** [comment]
Few comments, but thanks for the update 😄

File: src/kit/accessory.factory.js (hunk 2 above)
Comment (palourde) [reply src/kit/accessory.factory.js:8]: I actually had to define `session` as a variable [here](https://github.com/jfdoube/homebridge-sinope/pull/6/files#diff-5bf60c416b72e205d0c9a321a8a07b67R12) so I could re-use it when [logging out](https://github.com/jfdoube/homebridge-sinope/pull/6/files#diff-5bf60c416b72e205d0c9a321a8a07b67R80), but I'm sure there's a better way of doing it!

Edit: Could I simply do `{ session } = authData;`?

Comment (jfdoube) [reply src/kit/accessory.factory.js:8]: I think you get what you need with
```
const { session } = authData
```

**Review by jfdoube — APPROVED** [review]

**Comment by palourde** [comment]
Closing this PR as explained in https://github.com/jfdoube/homebridge-sinope/issues/5#issuecomment-731846978.

**Pull request closed without merging**